In [2]:
from testing_sample import Data_file,peak_detect,plot_gauss
import json
import numpy as np
import matplotlib.pyplot as plt
from scipy.signal import find_peaks, savgol_filter
from scipy.optimize import curve_fit
import math
import numba as nb



In [3]:
#aparature_data
ap = Data_file(-1)
ap.read_data()

In [4]:
index = []
with open("data_storage.json", "r", encoding="utf-8") as file:
    data = json.load(file)
    for i in range (len(data)):
        if 'Fe3O4' in  data[i]["name"] or 'fe3o4' in data[i]["name"]:
            if data[i]["data"]["hkl"] == False:
                continue
            else:
                index.append(i)
    print(len(index))

12


In [5]:
a,b,c,d,e,f,g,h,j,k,l,m = 0,0,0,0,0,0,0,0,0,0,0,0
names = [a,b,c,d,e,f,g,h,j,k,l,m]
print(len(names))
for i in range(len(index)):
    names[i] = Data_file(index[i])
    names[i].read_data()
    names[i].read_data_of_hkl()
    names[i].delete_aparature(ap.counts,ap.x,ap.step)


12


In [ ]:
# nr,eff,h,d,p = [],[],[],[],[]
# for i in range(len(names)):
#     nr_peaks,efficient,height,dist,prom = peak_detect(names[i].without_aparature_counts,names[i].x,11,45,0.01,names[i].two_theta,4,names[i].theta_start,names[i].step)
#     nr.append(nr_peaks)
#     eff.append(efficient)
#     h.append(height)
#     d.append(dist)
#     p.append(prom)
# print(nr)
# print(eff)
# print(h)
# print(d)
# print(p) 

[28, 41, 28, 41, 26, 39, 36, 28, 28, 28, 28, 17]
[0.6071428571428571, 0.47619047619047616, 0.6071428571428571, 0.47619047619047616, 0.6, 0.5, 0.6666666666666666, 0.6071428571428571, 0.6071428571428571, 0.3333333333333333, 0.3333333333333333, 0.4642857142857143]
[11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11, 11]
[45, 45, 45, 45, 45, 45, 45, 45, 45, 45, 45, 45]
[0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01, 0.01]


In [ ]:
# he = [4,6,8,10,20,25]
# di = [45,50,55]
# pr = [0.1,0.2,0.3]
# nr,eff,h,d,p = [],[],[],[],[]
# for i in range(len(names)):
#     for j in range(len(di)):
#         for k in range(len(pr)):
#             for l in range(len(he)):
#                 nr_peaks,efficient,height,dist,prom = peak_detect(names[i].without_aparature_counts,names[i].x,he[l],di[j],pr[k],names[i].two_theta,4,names[i].theta_start,names[i].step)
#                 nr.append(nr_peaks)
#                 eff.append(efficient)
#                 h.append(height)
#                 d.append(dist)
#                 p.append(prom)
# print(nr)
# print(eff)
# print(h)
# print(d)
# print(p)               
# # peak_detect(without_aparature_counts,he[0],40,0.2,f.two_theta)

# # peak_detect(without_aparature_counts,he[1],50,0.2,f.two_theta)
# # peak_detect(without_aparature_counts,he[2],50,0.2,f.two_theta)
# # peak_detect(without_aparature_counts,he[3],50,0.2,f.two_theta)
# # peak_detect(without_aparature_counts,hei[0],50,0.2,f.two_theta)
# # peak_detect(without_aparature_counts,hei[1],50,0.2,f.two_theta)
# # peak_detect(without_aparature_counts,hei[2],50,0.2,f.two_theta)
# # peak_detect(without_aparature_counts,hei[3],50,0.2,f.two_theta)
# #     l.append(li)
# #     r2.append(r2_)
# #     w.append(wy)
# # print(l)
# # print(r2)
# # print(w)
        

 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało się dopasować Gaussa do piku przy x = 41.34
 Nie udało

Znaleziono 27 pików. R^2 dopasowania: 0.9897,wydajność = 0.9642857142857143
Znaleziono 27 pików. R^2 dopasowania: 0.9897,wydajność = 0.5714285714285714,eps = 4
Znaleziono 16 pików. R^2 dopasowania: 0.9802,wydajność = 0.5714285714285714
Znaleziono 16 pików. R^2 dopasowania: 0.9802,wydajność = 0.5,eps = 4
Znaleziono 15 pików. R^2 dopasowania: 0.9800,wydajność = 0.5357142857142857
Znaleziono 15 pików. R^2 dopasowania: 0.9800,wydajność = 0.4642857142857143,eps = 4
Znaleziono 27 pików. R^2 dopasowania: 0.9897,wydajność = 0.9642857142857143
Znaleziono 27 pików. R^2 dopasowania: 0.9897,wydajność = 0.5714285714285714,eps = 4
Znaleziono 16 pików. R^2 dopasowania: 0.9802,wydajność = 0.5714285714285714
Znaleziono 16 pików. R^2 dopasowania: 0.9802,wydajność = 0.5,eps = 4
Znaleziono 15 pików. R^2 dopasowania: 0.9800,wydajność = 0.5357142857142857
Znaleziono 15 pików. R^2 dopasowania: 0.9800,wydajność = 0.4642857142857143,eps = 4

Znaleziono 16 pików. R^2 dopasowania: 0.9802,wydajność = 0.5, eps = 4, h = 20, d = 50, p = 0.2
Znaleziono 15 pików. R^2 dopasowania: 0.9800,wydajność = 0.4642857142857143, eps = 4, h = 25, d = 50, p = 0.2
Znaleziono 16 pików. R^2 dopasowania: 0.9802,wydajność = 0.5, eps = 4, h = 20, d = 50, p = 0.2
Znaleziono 15 pików. R^2 dopasowania: 0.9800,wydajność = 0.4642857142857143, eps = 4, h = 25, d = 50, p = 0.2
Znaleziono 16 pików. R^2 dopasowania: 0.9802,wydajność = 0.5, eps = 4, h = 20, d = 50, p = 0.2
Znaleziono 15 pików. R^2 dopasowania: 0.9800,wydajność = 0.4642857142857143, eps = 4, h = 25, d = 50, p = 0.2
Znaleziono 16 pików. R^2 dopasowania: 0.9802,wydajność = 0.5, eps = 4, h = 20, d = 50, p = 0.2
Znaleziono 15 pików. R^2 dopasowania: 0.9800,wydajność = 0.4642857142857143, eps = 4, h = 25, d = 50, p = 0.2
Znaleziono 16 pików. R^2 dopasowania: 0.9802,wydajność = 0.5, eps = 4, h = 20, d = 50, p = 0.2
Znaleziono 15 pików. R^2 dopasowania: 0.9800,wydajność = 0.4642857142857143, eps = 4, h = 25, d = 50, p = 0.2
Znaleziono 16 pików. R^2 dopasowania: 0.9802,wydajność = 0.5, eps = 4, h = 20, d = 50, p = 0.2
Znaleziono 15 pików. R^2 dopasowania: 0.9800,wydajność = 0.4642857142857143, eps = 4, h = 25, d = 50, p = 0.2
Znaleziono 16 pików. R^2 dopasowania: 0.9802,wydajność = 0.5, eps = 4, h = 20, d = 50, p = 0.2
Znaleziono 15 pików. R^2 dopasowania: 0.9800,wydajność = 0.4642857142857143, eps = 4, h = 25, d = 50, p = 0.2
Znaleziono 16 pików. R^2 dopasowania: 0.9802,wydajność = 0.5, eps = 4, h = 20, d = 50, p = 0.2
Znaleziono 15 pików. R^2 dopasowania: 0.9800,wydajność = 0.4642857142857143, eps = 4, h = 25, d = 50, p = 0.2
Znaleziono 16 pików. R^2 dopasowania: 0.9802,wydajność = 0.5, eps = 4, h = 20, d = 50, p = 0.2
Znaleziono 15 pików. R^2 dopasowania: 0.9800,wydajność = 0.4642857142857143, eps = 4, h = 25, d = 50, p = 0.2
Znaleziono 33 pików. R^2 dopasowania: 0.9713,wydajność = 0.42857142857142855, eps = 4, h = 20, d = 50, p = 0.2
Znaleziono 28 pików. R^2 dopasowania: 0.9680,wydajność = 0.47619047619047616, eps = 4, h = 25, d = 50, p = 0.2
Znaleziono 33 pików. R^2 dopasowania: 0.9713,wydajność = 0.42857142857142855, eps = 4, h = 20, d = 50, p = 0.2
Znaleziono 28 pików. R^2 dopasowania: 0.9680,wydajność = 0.47619047619047616, eps = 4, h = 25, d = 50, p = 0.2

In [15]:
max_value = max(eff)
max_index = eff.index(max_value)

print("Największa wartość:", max_value)
print("Indeks:", max_index)

Największa wartość: 0.6666666666666666
Indeks: 6


In [16]:
print(nr[max_index])
print(eff[max_index])
print(h[max_index])
print(d[max_index])
print(p[max_index]) 

print(nr[max_index-1])
print(eff[max_index-1])
print(h[max_index-1])
print(d[max_index-1])
print(p[max_index-1]) 

print(nr[max_index+1])
print(eff[max_index+1])
print(h[max_index+1])
print(d[max_index+1])
print(p[max_index+1]) 

36
0.6666666666666666
11
45
0.05
39
0.5
11
45
0.05
28
0.6071428571428571
11
45
0.05
